mongodbforms/fields.py

```diff
@@ -76,17 +76,18 @@ class ReferenceField(forms.ChoiceField):
     Reference field for mongo forms. Inspired by
     `django.forms.models.ModelChoiceField`.
     """
-    def __init__(self, queryset, empty_label="---------", *args, **kwargs):
+    def __init__(self, document, empty_label="---------", *args, **kwargs):
         forms.Field.__init__(self, *args, **kwargs)
         self.empty_label = empty_label
-        self.queryset = queryset
+        self._document = document
 
     def _get_queryset(self):
-        return self._queryset.clone()
+        return self._document.objects.clone()
 
     def _set_queryset(self, queryset):
-        self._queryset = queryset
+        # COMPAT: queryset always based on _document, but this setter is used for deepcopy()
```

Comment:
What about `deepcopy`? Worth a link to a doc.

Reply:
I initially removed the setter from the property, making it effectively read-only. The Django admin didn't like it: https://github.com/django/django/blob/1.10.5/django/forms/forms.py#L93

I'm assuming `deepcopy` handles `_document`, so I'm only keeping this setter around so it doesn't crash when copying the `queryset` attribute. And the weird `choices` handling...

Reply:
Ah nevermind, should have seen this bad boy.
https://github.com/Work4Labs/django-mongodbforms/blob/71ae55f78bae191a7d4b89be25be337e22fe7362/mongodbforms/fields.py#L130-L134

Reply:
Bwah?

Reply:
Deepcopy doesn't copy read-only properties, but this class defined a `__deepcopy__` implementation copying this attribute specifically. I've gotten rid of it.
4de9ab5 :white_check_mark: